server/src/main/java/keywhiz/service/daos/AclDAO.java

```diff
@@ -333,6 +333,34 @@ public void findAndEvictClient(long clientId, long groupId, AuditLog auditLog, S
         .map(row -> processSanitizedSecretRow(row, client));
   }
 
+  public List<SanitizedSecret> getBatchSanitizedSecretsFor(Client client, List<String> secretNames) {
```

Comment:
nit: It might be worth adding a test specifically for this endpoint, although the BatchSecretsIntegrationTest implicitly tests it.